### PR TITLE
431 Add environment variable options to Spacewalk

### LINKED ISSUE #431: Add environment variable options to Spacewalk
Can we add the parameters that spacewalk uses as environment variables as well? So the way it would work that spacewalk would take the environment variables instead of the command line parameters if there is nothing defined in the command line.

### PR DESCRIPTION
**Pull request by b-yap**
closes #431 

adding the ff. arguments an ENV option:
1. KEYRING
2. KEYFILE
3. KEYNAME
4. SPACEWALK_PARACHAIN_URL
5. STELLAR_VAULT_SECRET_KEY_FILEPATH
6. STELLAR_OVERLAY_CONFIG_FILEPATH
7. AUTO_REGISTER

note for **_AUTO_REGISTER_**:  since the value must be a tuple, make sure to **_enclose it with double quotes_**. 

:no_good: :x: `export AUTO_REGISTER=0,GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN:USDC,1000000` :x: :no_good:

:white_check_mark: :ok_hand: **`export AUTO_REGISTER="0,GAKNDFRRWA3RPWNLTI3G4EBSD3RGNZZOY5WKWYMQ6CQTG3KIEKPYWAYC:USDC,1000000"`** :ok_hand: :white_check_mark:

For arguments that accept _`option`_,  make sure to **_enclose it with double quotes_** with the _ACTUAL_ value.
:no_good: :x: `export MAX_NOTIFS_PER_SUBSCRIPTION="1"` :x: :no_good:
:no_good: :x: `export MAX_NOTIFS_PER_SUBSCRIPTION="Some(1)"` :x: :no_good:

:white_check_mark: :ok_hand: **`export MAX_NOTIFS_PER_SUBSCRIPTION=1`** :ok_hand: :white_check_mark:

### CODE REVIEW

**Review by TorstenStueber — APPROVED** [review]
Looks good.

**Review by ebma — COMMENTED** [review]
Looks good, but is there a reason not to make all configuration parameters available as environment variables? We don't really use the ones that you didn't add but they could be useful for other users.

**Comment by b-yap** [comment]
@ebma I added env option for ALL args.

File: clients/runtime/src/cli.rs [hunk 1]
```diff
@@ -93,15 +93,15 @@ pub struct ConnectionOpts {
 	pub spacewalk_parachain_url: String,
 
 	/// Timeout in milliseconds to wait for connection to spacewalk-parachain.
-	#[clap(long, parse(try_from_str = parse_duration_ms), default_value = "60000")]
+	#[clap(long, env = "SPACEWALK_PARACHAIN_CONN_TIMEOUT_MS", parse(try_from_str = parse_duration_ms), default_value = "60000")]
```
Comment (ebma) [inline clients/runtime/src/cli.rs:96]:
```suggestion
	#[clap(long, env = "SPACEWALK_PARACHAIN_CONNECTION_TIMEOUT_MS", parse(try_from_str = parse_duration_ms), default_value = "60000")]
```

File: clients/runtime/src/cli.rs [hunk 2]
```diff
@@ -93,15 +93,15 @@ pub struct ConnectionOpts {
...
 	pub spacewalk_parachain_connection_timeout_ms: Duration,
 
 	/// Maximum number of concurrent requests
-	#[clap(long)]
+	#[clap(long, env = "MAX_CONCURRENT_REQUESTS")]
 	pub max_concurrent_requests: Option<usize>,
 
 	/// Maximum notification capacity for each subscription
-	#[clap(long)]
+	#[clap(long, env = "S")]
```
Comment (ebma) [inline clients/runtime/src/cli.rs]:
```suggestion
	#[clap(long, env = "MAX_NOTFIS_PER_SUBSCRIPTION")]
```

**Review by ebma — COMMENTED** [review]
Since I guess there is no way to spit out all available names of environment variables from the vault binary, I think it's best to give each env variable the very same name as the config item. Then it's at least somewhat obvious which names are used/available.

File: clients/runtime/src/cli.rs (hunk 1 above)
Comment (b-yap) [reply clients/runtime/src/cli.rs]: I use `--help` to show all args.

**Review by ebma — APPROVED** [review]

**Pull request merged**